pandas/tests/frame/test_reshape.py

```diff
@@ -1128,3 +1128,21 @@ def test_stack_timezone_aware_values():
         ),
     )
     tm.assert_series_equal(result, expected)
+
+
+def test_unstacking_multi_index_df():
+    # BUG: gh-30740
+    df = pd.DataFrame(
+        {
+            "name": ["Alice", "Bob"],
+            "score": [9.5, 8],
+            "employed": [False, True],
+            "kids": [0, 0],
+            "gender": ["female", "male"],
+        }
+    )
+    df = df.set_index(["name", "employed", "kids", "gender"])
+    df = df.unstack(["gender"], fill_value=0)
+    expected = df.unstack("employed", fill_value=0).unstack("kids", fill_value=0)
```

Comment:
I'd feel slightly more comfortable if this were constructed directly, rather than via unstack. If you have time to update that, then great. Otherwise, I'll merge before tagging the release in a few hours.

Reply:
@TomAugspurger 

```
expected = pd.DataFrame(
    [[9.5, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 8.0]],
    pd.Index(["Alice", "Bob"], name="name"),
    columns=pd.MultiIndex.from_tuples(
        [
            ("score", "female", False, 0),
            ("score", "female", True, 0),
            ("score", "male", False, 0),
            ("score", "male", True, 0),
        ],
        names=[None, "gender", "employed", "kids"],
    ),
)
```

would this be ok?

Reply:
don't need the pds though.

Reply:
I can change that. What is meant by pds ?

Reply:
Index, MultiIndex and DataFrame are imported. so adding the pd prefix isn't necessary.

Reply:
Ok. Got it. Thanks

Reply:
I also missed `index=` so should be

```
    expected = DataFrame(
        [[9.5, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 8.0]],
        index=Index(["Alice", "Bob"], name="name"),
        columns=MultiIndex.from_tuples(
            [
                ("score", "female", False, 0),
                ("score", "female", True, 0),
                ("score", "male", False, 0),
                ("score", "male", True, 0),
            ],
            names=[None, "gender", "employed", "kids"],
        ),
    )
```

Reply:
Added this